src/Microsoft.AspNet.Routing/Resources.resx

```diff
@@ -189,4 +186,10 @@
   <data name="TemplateRoute_UnescapedBrace" xml:space="preserve">
     <value>In a route parameter, '{' and '}' must be escaped with '{{' and '}}'</value>
   </data>
+  <data name="TemplateRoute_OptionalParameterCanbBePrecededByPeriod" xml:space="preserve">
+    <value>In the segment '{0}', the optional parameter '{1}' is preceded by an invalid segment '{2}'. Only valid literal to precede an optional parameter is a period (.).</value>
+  </data>
```

Comment:
This is not fixed, `Only valid literal to precede an optional parameter is a period (.)` is not a complete sentence